### PR TITLE
Issue#1021

### PR DESCRIPTION
**Pull request by balif**
Issue #1021 fix

### CODE REVIEW

File: service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java [hunk 1]
```diff
@@ -172,6 +173,7 @@ public Uni<List<ServiceInstance>> fetchNewServiceInstances(List<ServiceInstance>
                             LOGGER.error("Unable to retrieve the endpoint from the {} service", application,
                                     result.cause());
                             emitter.fail(result.cause());
+                            invalidated.set(true);
```
Comment (aureamunoz) [inline service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java]: Setting invalidated = true after an error may cause an infinite loop of failed attempts if the error persists. Wasn't this the original issue? We would also need a test to reproducing the problem and validating the corresponding fix.

File: service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java [hunk 2]
```diff
@@ -127,6 +127,7 @@ public void invalidate() {
 
     @Override
     public Uni<List<ServiceInstance>> fetchNewServiceInstances(List<ServiceInstance> previousInstances) {
+        invalidated.set(false);
```
Comment (aureamunoz) [inline service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java:130]: Ideally, it should be modified only after success or confirmed failure.

File: service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java (hunk 1 above)
Comment (balif) [reply service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java]: in my case it didn't notice any problems with queries to k8s api. I added this line as an idea how to make it so that in case of an error another query to api will be sent. In fact it can cause the original error to return which was fixed for me by adding the line: 130

File: service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java (hunk 2 above)
Comment (balif) [reply service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java:130]: @aureamunoz I have added test. 
After Removing invalidated.set(false); test will fail

Comment (aureamunoz) [reply service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java:130]: Thank you. I am doing some tests on my side to see what is the best solution to cover the corner cases as well. I'll let you know
